### PR TITLE
[Doc] Improve quickstart documentation

### PR DESCRIPTION
**Pull request by rafvasq**
This PR tries to improve the existing quickstart guide for newcomers to the project who want to generate something with vLLM quickly. 

- Include pre-req system requirements 
- Include installation step
- Reorganize sections
- Add links to help with LLM terminology
- Improve step-by-step guide readability
- Use [Qwen2.5-1.5B-Instruct](https://huggingface.co/Qwen/Qwen2.5-1.5B-Instruct) for chat/server sections

### CODE REVIEW

File: docs/source/getting_started/quickstart.rst [hunk 1]
```diff
@@ -44,59 +51,61 @@ Define the list of input prompts and the sampling parameters for generation. The
...
-By default, it starts the server at ``http://localhost:8000``. You can specify the address with ``--host`` and ``--port`` arguments. The server currently hosts one model at a time (OPT-125M in the command below) and implements `list models <https://platform.openai.com/docs/api-reference/models/list>`_, `create chat completion <https://platform.openai.com/docs/api-reference/chat/completions/create>`_, and `create completion <https://platform.openai.com/docs/api-reference/completions/create>`_ endpoints. We are actively adding support for more endpoints.
+By default, it starts the server at ``http://localhost:8000``. You can specify the address with ``--host`` and ``--port`` arguments. The server currently hosts one model at a time and implements such as `list models <https://platform.openai.com/docs/api-reference/models/list>`_, `create chat completion <https://platform.openai.com/docs/api-reference/chat/completions/create>`_, and `create completion <https://platform.openai.com/docs/api-reference/completions/create>`_ endpoints. 
 
-Start the server:
+To start the server, simply use:
 
-.. code-block:: console
+.. code-block:: shell
```
Comment (ckadner) [inline docs/source/getting_started/quickstart.rst:87]: I don't see `shell` listed here: https://build-me-the-docs-please.readthedocs.io/en/latest/Using_Sphinx/ShowingCodeExamplesInSphinx.html#pygments-lexers, only `bash` or `console`, where the latter appears to be more generic

File: docs/source/getting_started/quickstart.rst [hunk 2]
```diff
@@ -44,59 +51,61 @@ Define the list of input prompts and the sampling parameters for generation. The
...
 
     $ curl http://localhost:8000/v1/models
 
 You can pass in the argument ``--api-key`` or environment variable ``VLLM_API_KEY`` to enable the server to check for API key in the header.
 
-Using OpenAI Completions API with vLLM
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+OpenAI Completions API with vLLM
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (ckadner) [inline docs/source/getting_started/quickstart.rst:104]:
```suggestion
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: docs/source/getting_started/quickstart.rst [hunk 3]
```diff
@@ -124,18 +133,16 @@ Since this server is compatible with OpenAI API, you can use it as a drop-in rep
...
-Using OpenAI Chat API with vLLM
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+A more detailed client example can be found `here <https://github.com/vllm-project/vllm/blob/main/examples/openai_completion_client.py>`__.
 
-The vLLM server is designed to support the OpenAI Chat API, allowing you to engage in dynamic conversations with the model. The chat interface is a more interactive way to communicate with the model, allowing back-and-forth exchanges that can be stored in the chat history. This is useful for tasks that require context or more detailed explanations.
+OpenAI Chat API with vLLM
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Querying the model using OpenAI Chat API:
+vLLM is also designed to support the OpenAI Chat API. The chat interface is a more dynamic, interactive way to communicate with the model, allowing back-and-forth exchanges that can be stored in the chat history. This is useful for tasks that require context or more detailed explanations.
```
Comment (ckadner) [inline docs/source/getting_started/quickstart.rst:141]:
```suggestion
vLLM is designed to also support the OpenAI Chat API. The chat interface is a more dynamic, interactive way to communicate with the model, allowing back-and-forth exchanges that can be stored in the chat history. This is useful for tasks that require context or more detailed explanations.
```

File: docs/source/getting_started/quickstart.rst [hunk 4]
```diff
@@ -124,18 +133,16 @@ Since this server is compatible with OpenAI API, you can use it as a drop-in rep
...
 
-You can use the `create chat completion <https://platform.openai.com/docs/api-reference/chat/completions/create>`_ endpoint to communicate with the model in a chat-like interface:
+You can use the `create chat completion <https://platform.openai.com/docs/api-reference/chat/completions/create>`_ endpoint:
```
Comment (ckadner) [inline docs/source/getting_started/quickstart.rst:143]:
```suggestion
You can use the `create chat completion <https://platform.openai.com/docs/api-reference/chat/completions/create>`_ endpoint to interact with the model:
```

**Review by ckadner — COMMENTED** [review]
Thank you @rafvasq for improving the quickstart guide. I have a few nits and some questions. I'm new rst formatting as and some of my suggestion are likely just personal preference. Feel free to ignore those :-)

File: docs/source/getting_started/quickstart.rst [hunk 5]
```diff
@@ -1,38 +1,45 @@
 .. _quickstart:
 
+==========
 Quickstart
 ==========
 
-This guide shows how to use vLLM to:
+This guide will help you quickly get started with vLLM to:
```
[the remark this answers is not in the document]
Comment (rafvasq) [reply docs/source/getting_started/quickstart.rst:7]: Changed them to use the refs

**Comment by rafvasq** [comment]
Thanks @ckadner, I believe I addressed everything!

**Review by ckadner — APPROVED** [review]
Looks good. Thanks @rafvasq

**Comment by simon-mo** [comment]
This looks great! Can you also update the chat/server section to ues https://huggingface.co/Qwen/Qwen2.5-1.5B-Instruct so that the users don't need a chat template to start with (advanced feature)?

**Comment by rafvasq** [comment]
> ! Can you also update the chat/server section to ues https://huggingface.co/Qwen/Qwen2.5-1.5B-Instruct so that the users don't need a chat template to start with (advanced feature)?

Hey @simon-mo, would removing the extra note about chat templates suffice? It doesn't seem to me like a user needs to know anything about chat templates to use the guide with facebook/opt-125m as written since vLLM just uses a default quietly.

**Comment by simon-mo** [comment]
ah that's a good point. however my main problem with the opt model is that just generates gibberish for chat.

**Comment by rafvasq** [comment]
Makes sense, I've updated the model for those sections

**Review by simon-mo — APPROVED** [review]

**Pull request merged**